src/kernel/sporadic.c

```diff
@@ -249,100 +237,57 @@ static bool_t refill_head_overlapping(sched_context_t *sc)
 void refill_budget_check(ticks_t usage)
```

Comment:
Are you able to add to the commit message how the algorithm for processing the charging of the given usage has been changed also? Currently the commit message implies that this change is only modifying the structure of the code and doesn't imply that the behavior has also changed.  I can see that the way overrun is processed has changed, but can't easily see if anything else has changed also.

Reply:
* [ ] Fix commit messages to reflect changes before merging

Reply:
Are you able to quickly comment what changed here to make it easier for me to review?

Reply:
The algorithm for dealing with charging a budget needed to be fundamentally the same regardless of whether it was being charged due to preemption (the 'split check') or due to exhaustion/job completion (the original 'budget check'). I suspect this is due to invariants that we still need to enforce in both cases (that the head refill has MIN_BUDGET) that cause these to both effectively behave the same.

The process to charge any time to an SC is to charge that time to the head refill, and append a new refill one period in the future (which is merged if it ends up adjacent to or overlapping an existing refill). The head refill is then moved to ensure that the head and tail refills bound a 1-period window. We then delay and merge refills at the head until the head refill has sufficient budget and is not overlapping any other refills.